src/main/java/io/cdap/plugin/WindowAggregation.java

```diff
@@ -240,24 +303,169 @@ private String generateFrameDescription(WindowAggregationConfig config) {
         return ".";
       case ROW:
         return String.format("over a windows frame consisting of %s preceding and %s following rows.",
-                             isUnboundedPreceding ? "unbound" : config.getFrameDefinitionPrecedingBound(),
-                             isUnboundedFollowing ? "unbound" : config.getFrameDefinitionFollowingBound());
+          isUnboundedPreceding ? "unbound" : config.getFrameDefinitionPrecedingBound(),
+          isUnboundedFollowing ? "unbound" : config.getFrameDefinitionFollowingBound());
       case RANGE:
         return String.format("over a windows frame consisting of %s preceding and %s following range.",
-                             isUnboundedPreceding ? "unbound" : config.getFrameDefinitionPrecedingBound(),
-                             isUnboundedFollowing ? "unbound" : config.getFrameDefinitionFollowingBound());
+          isUnboundedPreceding ? "unbound" : config.getFrameDefinitionPrecedingBound(),
+          isUnboundedFollowing ? "unbound" : config.getFrameDefinitionFollowingBound());
     }
 
     return "";
   }
 
   @Override
-  public JavaRDD<StructuredRecord> transform(
-    SparkExecutionPluginContext sparkExecutionPluginContext, JavaRDD<StructuredRecord> javaRDD) throws Exception {
+  public JavaRDD<StructuredRecord> transform(SparkExecutionPluginContext sparkExecutionPluginContext,
+     JavaRDD<StructuredRecord> javaRDD) throws Exception {
     Schema inputSchema = sparkExecutionPluginContext.getInputSchema();
     if (inputSchema == null) {
       throw new Exception("Input schema is null. Input schema can not be null at this stage");
     }
     return WindowsAggregationUtil.transform(sparkExecutionPluginContext, javaRDD, config, inputSchema, outputSchema);
   }
+
+  @Override
+  public boolean canUseEngine(Engine engine) {
+    Optional<ExpressionFactory<String>> expressionFactory = engine.
+      getExpressionFactory(StringExpressionFactoryType.SQL);
+    return expressionFactory.isPresent();
+  }
+
+  //Generate window aggregation definition by setting values
+  @Nullable
+  public WindowAggregationDefinition generateAggregationDefinition(RelationalTranformContext ctx, Relation relation,
+    FailureCollector failureCollector, ExpressionFactory<String> expressionFactory) {
+    WindowAggregationConfig.WindowFrameType windowFrameType = config.getWindowFrameType();
+    WindowAggregationDefinition.WindowFrameType windowFrameTypeValue = null;
+    boolean unboundedPreceding = false;
+    boolean unboundedFollowing = false;
+    String preceding = null;
+    String following = null;
+    if (windowFrameType != WindowAggregationConfig.WindowFrameType.NONE) {
```

Comment:
nit: Could you please add an empty line above this line to delineate variable declarations?

Reply:
Done